flink-streaming-java/src/main/java/org/apache/flink/streaming/api/graph/StreamGraphGenerator.java

```diff
@@ -385,7 +335,9 @@ private void configureStreamGraph(final StreamGraph graph) {
 
         if (shouldExecuteInBatchMode) {
             configureStreamGraphBatch(graph);
-            setDefaultBufferTimeout(-1);
+            configuration.set(
+                    ExecutionOptions.BUFFER_TIMEOUT,
+                    Duration.ofMillis(ExecutionOptions.DISABLED_NETWORK_BUFFER_TIMEOUT));
```

Comment:
If we consider `BUFFER_TIMEOUT_ENABLED` when getBufferTimeout, it's better to set `BUFFER_TIMEOUT_ENABLED` to false here.

Reply:
ditto